go/flexver/main_test.go

```diff
@@ -9,219 +9,181 @@
 	}
 }
 
-func TestLess(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			if v.op == shouldError {
-				// Catch panic
-				defer func() { _ = recover() }()
-			}
+func TestStandardized(t *testing.T) {
+	for _, test := range ENABLED_TESTS {
+		file, err := os.Open("../../test/" + test)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer file.Close()
 
-			res := Less(v.a, v.b)
-			if (v.op == opLT && res == false) ||
-				(v.op == opEQ && res == true) ||
-				(v.op == opGT && res == true) ||
-				(v.op == shouldError) {
-				t.Fatalf("Unexpected result %v", res)
-			}
-		})
-	}
-}
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			line := scanner.Text()
 
-func TestEqual(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			if v.op == shouldError {
-				// Catch panic
-				defer func() { _ = recover() }()
+			if strings.HasPrefix(line, "#") {
+				continue
+			}
+			if len(line) == 0 {
+				continue
 			}
 
-			res := Equal(v.a, v.b)
-			if (v.op == opLT && res == true) ||
-				(v.op == opEQ && res == false) ||
-				(v.op == opGT && res == true) ||
-				(v.op == shouldError) {
-				t.Fatalf("Unexpected result %v", res)
+			split := strings.Split(line, " ")
+			if len(split) != 3 {
+				t.Fatal("Line formatted incorrectly, expected 2 spaces: " + line)
 			}
-		})
-	}
-}
 
-func TestCompareError(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			res, err := CompareError(v.a, v.b)
-			if err == nil {
-				if (v.op == opLT && res >= 0) ||
-					(v.op == opEQ && res != 0) ||
-					(v.op == opGT && res <= 0) ||
-					(v.op == shouldError) {
-					t.Fatalf("Unexpected result %v", res)
-				}
-			} else if v.op != shouldError {
-				t.Fatalf("Unexpected error %v", err)
+			ord := 0
+			switch split[1] {
+			case "<":
+				ord = opLT
+			case "=":
+				ord = opEQ
+			case ">":
+				ord = opGT
 			}
-		})
+
+			lefthand := split[0]
+			righthand := split[2]
+
+			t.Run(line, func(t *testing.T) {
+				t.Run("Compare", func(t *testing.T) {
+					RunCompare(t, lefthand, righthand, ord)
+				})
+
+				t.Run("Less", func(t *testing.T) {
+					RunLess(t, lefthand, righthand, ord)
+				})
+
+				t.Run("Equal", func(t *testing.T) {
+					RunEqual(t, lefthand, righthand, ord)
+				})
+			})
+		}
+
+		if err := scanner.Err(); err != nil {
+			t.Fatal(err)
+		}
 	}
 }
-
-func TestLessError(t *testing.T) {
-	for _, v := range allTests {
-		t.Run(v.name(), func(t *testing.T) {
-			res, err := LessError(v.a, v.b)
-			if err == nil {
-				if (v.op == opLT && res == false) ||
-					(v.op == opEQ && res == true) ||
-					(v.op == opGT && res == true) ||
-					(v.op == shouldError) {
-					t.Fatalf("Unexpected result %v", res)
-				}
-			} else if v.op != shouldError {
-				t.Fatalf("Unexpected error %v", err)
-			}
-		})
+func TestInvalid(t *testing.T) {
+	_, err := CompareError("\xc3\x28", "")
```

Comment:
This no longer tests whether the panicking variants panic on invalid input; `defer func() { _ = recover() }()` can be used to catch a panic as done in the previous implementation.